src/robotFactory.js

```diff
@@ -1,10 +1,115 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion,
+  ) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if (coords.x || coords.y) {
+      this.coords = coords;
+    } else {
+      this.coords = {
+        x: 0,
+        y: 0,
+      };
+    }
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  getInfo() {
+    const name = `Robot: ${this.name}`;
+    const version = `Chip version: ${this.chipVersion}`;
+    const weight = `Weight: ${this.weight}`;
+
+    return `${name}, ${version}, ${weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion,
+  ) {
+    super(
+      name,
+      weight,
+      coords,
+      chipVersion,
+    );
```

Comment:
you can use arguments
```suggestion
    super(...arguments);
```